src/main/java/org/fenixedu/academic/util/AcademicPhoneNumberHandler.java

```diff
@@ -0,0 +1,119 @@
+        DEFAULT_RESOLVERS.put(Phone.class, (pc) -> ((Phone) pc).getNumber());
+        DEFAULT_RESOLVERS.put(PhysicalAddress.class, (pc) -> ((PhysicalAddress) pc).getAddress());
+        DEFAULT_RESOLVERS.put(WebAddress.class, (pc) -> ((WebAddress) pc).getUrl());
+    }
+
+    private static final Collection<PhoneNumberType> FIXED_NUMBERS;
+    private static final Collection<PhoneNumberType> MOBILE_NUMBERS;
+
+    static {
+        FIXED_NUMBERS = new ArrayList<PhoneNumberType>();
+        FIXED_NUMBERS.add(PhoneNumberType.VOIP);
+        FIXED_NUMBERS.add(PhoneNumberType.FIXED_LINE);
+
+        MOBILE_NUMBERS = new ArrayList<PhoneNumberType>();
+        MOBILE_NUMBERS.add(PhoneNumberType.MOBILE);
+    }
+
+    private static ContactResolver getResolver(Class<? extends PartyContact> class1) {
+        synchronized (DEFAULT_RESOLVERS) {
+            return DEFAULT_RESOLVERS.get(class1);
+        }
+    }
+
+    private PhoneNumberType getPhoneNumberType(PhoneNumber phoneNumber) {
+        return phoneNumber != null ? PHONE_UTIL.getNumberType(phoneNumber) : null;
+    }
+
+    @Override
+    public boolean isMobileNumber(String numberText) {
+        return isType(getPhoneNumberType(parsePhoneNumber(numberText)), MOBILE_NUMBERS);
+    }
+
+    @Override
+    public boolean isFixedNumber(String numberText) {
+        return isType(getPhoneNumberType(parsePhoneNumber(numberText)), FIXED_NUMBERS);
+    }
+
+    private boolean isType(PhoneNumberType type, Collection<PhoneNumberType> types) {
+        return types.contains(type);
+    }
+
+    @Override
+    public boolean shouldReceiveValidationSMS(String number) {
+        return isMobileNumber(number);
+    }
+
+    @Override
+    public boolean shouldReceiveValidationCall(String number) {
+        return isFixedNumber(number);
+    }
+
+    @Override
+    public PhoneNumber parsePhoneNumber(String numberText) {
+        if (!StringUtils.isEmpty(numberText)) {
+
+            if (numberText.startsWith("00")) {
+                numberText = numberText.replaceFirst("00", "+");
+            }
+
+            try {
+                final PhoneNumber phoneNumber = PHONE_UTIL.parse(numberText, null);
+                if (PHONE_UTIL.isValidNumber(phoneNumber)) {
+                    return phoneNumber;
+                }
+            } catch (NumberParseException e) {
+                System.out.println("O n�mero n�o � v�lido:" + e);
+                return null;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getInternationalFormatNumber(String numberText) {
+        final PhoneNumber phoneNumber = parsePhoneNumber(numberText);
+        if (phoneNumber != null) {
+            return PHONE_UTIL.format(phoneNumber, PhoneNumberFormat.E164);
+        }
+
+        return null;
+    }
+
+    @Override
+    public String resolve(PartyContact contact) {
+        return getResolver(contact.getClass()).getPresentationValue(contact);
```

Comment:
Calling `getResolver(contact.getClass())` may result in a null return value.
This should be checked because we never really know what the underlying map contains.